### PR TITLE
Added MPC and Linear MPC examples

### PR DESCRIPTION
**Pull request by adamhall**
As pet title.  Also added a `pytest` for mpc and linear mpc. Gains could be better, especially for quad stabilization, but they all work reasonably well.

### CODE REVIEW

File: examples/mpc/mpc_experiment.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# LQR Experiment.
```
Comment (Federico-PizarroBejarano) [inline examples/mpc/mpc_experiment.sh:3]: says LQR experiment lol

File: examples/mpc/mpc_experiment.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
+'''A LQR and iLQR example.'''
```
Comment (Federico-PizarroBejarano) [inline examples/mpc/mpc_experiment.py:1]: Also says LQR

File: examples/mpc/mpc_experiment.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+from matplotlib.ticker import FormatStrFormatter
+
+from safe_control_gym.envs.benchmark_env import Task
+from safe_control_gym.experiments.base_experiment import BaseExperiment
+from safe_control_gym.utils.configuration import ConfigFactory
+from safe_control_gym.utils.registration import make
+
+
+def run(gui=True, n_episodes=1, n_steps=None, save_data=False):
+    '''The main function running LQR and iLQR experiments.
```
Comment (Federico-PizarroBejarano) [inline examples/mpc/mpc_experiment.py:19]: Also says LQR

**Comment by adamhall** [comment]
Fixed Typos

**Pull request merged**